src/certif/certifChecker.ml

```diff
@@ -3075,7 +3060,45 @@ let generate_all_proofs uid input sys =
 
         begin match Sys.command cmd with
           | 0 | 20 ->
-            files_cat_open [inv_lfsc; front_lfsc] final_lfsc |> Unix.close;
+            let read_lines filename =
+              let f = open_in filename in
+              let rec loop () =
+                try
+                  let next = input_line f in
+                  next :: loop ()
+                with End_of_file ->
+                  close_in f;
+                  []
+              in
+              loop ()
+            in
+            let rec write_lines oc = function
+              | [] -> ()
+              | l :: ls ->
+                  Printf.fprintf oc "%s\n" l;
+                  write_lines oc ls
+            in
+            let sym_defs =
+              List.map
+                (fun sym -> "(define " ^ HString.string_of_hstring sym)
+                !syms
+            in
+            let front_lines = read_lines front_lfsc in
+            let pred l =
+              List.exists (fun sym -> Lib.string_starts_with l sym) sym_defs
+              |> not
+            in
+            let front_lines = List.filter pred front_lines in
+            file_copy inv_lfsc final_lfsc;
+            let oc =
+              open_out_gen [ Open_append; Open_creat ] 0o666 final_lfsc
+            in
+            write_lines oc front_lines;
```

Comment:
Since LFSC proofs can be very big, I suggest replacing:
```ocaml
write_lines oc front_lines;
```
This would make 
with
```ocaml
filter_and_copy_lines (open_in front_lfsc) oc pred ;
```
where `filter_and_copy_lines` is:
```ocaml
let filter_and_copy_lines ic oc keep =
  let rec read_line () =
    try
      let line = input_line ic in
      if keep line then
        Printf.fprintf oc "%s\n" line ;
      read_line ()
    with End_of_file -> ()
  in
  read_line ()
in
```
This would make `read_lines` and `write_lines` unnecessary.